### ISSUE TITLE
Introduce Hypothesis for property-based testing of numerical invariants

### ISSUE THREAD

**Issue by xcz0** [xcz0/Eco-AutoDrive#100]
## Motivation

The repository already uses `pytest` and keeps fast CPU tests separate from `gpu`, `simulator`, and `slow` tests. Several core modules are especially suitable for property-based testing because they expose mathematical invariants and boundary-sensitive numerical behavior.

Hypothesis should complement the existing worked examples and regression tests rather than replace them. Existing explicit-value tests remain important for research semantics, units, reward formulas, episode boundaries, and reproducibility.

## Proposed scope

Add `hypothesis` to the development dependencies and introduce a small first batch of high-value property tests in the fast CPU suite.

Prioritize:

- `envs/domain/geometry.py`
  - local/world coordinate round trips
  - rotation preserving distances/norms
  - periodicity and output range of `shortest_angle_delta`
  - rear-axle geometry invariants
- `envs/domain/trajectory.py`
  - shape and finiteness invariants
  - heading wrapping / angular velocity behavior
  - consistency of derived velocities with finite differences
- reward components
  - scores always remain in `[0, 1]`
  - monotonicity where expected
  - correct saturation at configured thresholds
  - stationary/negative-progress edge cases
- reward configuration validation
  - relational constraints such as `safe_ttc > critical_ttc`
  - `maximum_ttc >= safe_ttc`
  - speed threshold ordering
  - calibrated energy-band ordering

## Non-goals

Do not use Hypothesis initially for:

- real MetaDrive simulator loops
- GPU tests
- long-running checkpoints/training workflows
- expensive end-to-end PPO optimizer loops

These cases are too costly or nondeterministic for broad generated-input exploration and should retain their current targeted tests.

## Testing strategy

Property tests should be additive. Keep the existing concrete regression/worked-example tests for known formulas and scientifically meaningful cases.

Prefer bounded finite strategies that reflect the valid numerical domain rather than unrestricted IEEE-754 values. Use small array/tensor sizes and keep default execution fast enough for `just test-all-cpu`.

A representative geometry property:

```python
@given(...)
def test_world_local_transform_round_trip(points, anchor, heading):
    world = local_points_to_world(points, anchor, heading)
    recovered = world_points_to_local(world, anchor, heading)
    np.testing.assert_allclose(recovered, points, rtol=1e-10, atol=1e-8)
```

## Acceptance criteria

- [ ] Add `hypothesis` to the dev dependency group.
- [ ] Add an initial set of roughly 10–15 focused property tests.
- [ ] Cover geometry/angle invariants and at least one reward monotonicity or saturation invariant.
- [ ] Cover at least one relational configuration constraint with generated valid/invalid inputs.
- [ ] Keep the tests in the default CPU suite; no `gpu`, `simulator`, or `slow` requirement.
- [ ] Existing explicit regression/worked-example tests remain intact unless a specific duplicate is clearly redundant.
- [ ] `just test-all-cpu` remains reasonably fast and deterministic.

## Follow-up

If the first batch proves useful, consider extending property tests to compact rollout/GAE data contracts (shape, terminal/bootstrap, finite outputs) while still avoiding expensive full training loops.
